### ISSUE TITLE
disable hook seems to have wrong true/false logic

### ISSUE THREAD

**Issue by beeplin** [feathersjs/feathers-hooks-common#28]
[Code here](https://github.com/feathersjs/feathers-hooks-common/blob/master/src/bundled.js#L336) means to disable when the result is false, not to disable when the result is true.
```js
if (typeof realm === 'function') {
    return function (hook) {
      const result = realm(hook);
      const update = check => {
        if (!check) {
          throw new errors.MethodNotAllowed(`Calling '${hook.method}' not allowed. (disable)`);
        }
      };
  }
```
But in [docs here](https://docs.feathersjs.com/hooks/bundled.html) it seems to be opposite logic: 

```js
  // Disable the remove hook if the user is not an admin
  remove: hooks.disable(function(hook) {
    return !hook.params.user.isAdmin
  })
```

Either the doc or the code is wrong. I guest it's the code which is wrong.

**Comment by eddyystop** [comment]
Thanks for your example. I assume it should be `if (check) {` but let me ask what the ramifications might be regarding changing the code.

As an aside, a number of hooks, like disable, allow their last param to be a function. This function indicates if the hook should be run or not. That feature will be deprecated in all these hooks and removed in the future.

I recommend you use `hooksCommon.iff(condition, hook)` where condition may be sync, or async using a Promise. `iff(condition, hook1, hook2, ...)` will be supported in the future.

**Comment by ekryski** [comment]
@eddyystop if it is wrong it's a bug and likely should be a patch release to feathers-hooks. However since it might cause a lot of breakage for people we **may** want to do a feathers-hooks major release. What do you think @daffl?

Regarding a release to `feathers-hooks-common` I think we can be safe to do a patch release as I don't know how many people have moved over, **or** we just follow the same process as with `feathers-hooks`.

**Comment by eddyystop** [comment]
feathers-hooks now exports feathers-hooks-common as its bundled hooks. So
there is no moving over as such.

**Comment by daffl** [comment]
We should probably just fix the docs then.

**Comment by ekryski** [comment]
If that bug has been in there a while and it is now used by feathers-hooks there are likely people that might be depending on that functionality. I'm surprised it wasn't caught earlier...

The question is, should we fix this and issue a major release? I think so. Even though it is a bug and typically a patch release, this really is the wrong logic and would be a breaking change for people.

So I think the course of action is:

- As @daffl suggested we should update the docs to at least use the new recommended syntax.
- Fix and issue a major release for feathers-hooks-common
- Possibly update feathers-hooks to use the new version of feathers-hooks-common and do a major release, or just leave it and do the major at a later date when we have other breaking changes.

**Comment by daffl** [comment]
I'm saying the docs should document the way it is implemented right now. I don't think it's worth making another major release of both this repository and feathers-hooks to fix a bug that apparently has gone unnoticed for a long time if we can just document the wrong way and fix it later (e.g. when hooks are part of core) and feathers-hooks-commons stands on its own.

**Comment by ekryski** [comment]
Releases are pretty cheap... I think we should just do the right thing, document what it should be and do the major releases. The only barrier to a major release is a mental one. It's the same process.

Otherwise until stuff is in core we'll run the risk of people not knowing what the actual right way is. The reason we haven't run into this issue is because we never pass callback functions and instead wrap hooks or just use `hooks.disable('external')`.

**Comment by daffl** [comment]
Making a major release is much more than just running `npm release:major`. You have to notify people about the new release (not everybody is using Greenkeeper), provide documentation for the migration and update all existing documentation, examples, generators etc.

If we have to make a major feathers-hooks release we might as well remove the bundled hooks completely.

**Comment by eddyystop** [comment]
@daffl, @ekryski what do you think of creating a new hook with the correct sense? Any name suggestions?

**Comment by eddyystop** [comment]
*ping*

@daffl, @ekryski what do you think of creating a new hook with the correct sense? Any name suggestions?

**Comment by daffl** [comment]
Couldn't this just be wrapped into an `iff` (which I'd still like to call `when` 😉)?

**Comment by eddyystop** [comment]
Any suggestions for the name of the hook with the opposite sense?

`iff` has been in use in the wild for some months now. Perhaps that boat
has sailed.

**Comment by ekryski** [comment]
I had suggested on a different issue that `when` could be an alias for `iff` and `unless` could be an alias for `iff(isNot())`.

**Comment by ekryski** [comment]
but in any case, none of that really has any bearing on the original issue. I agree with @eddyystop that you should probably just use `iff` to correct the logic, I've been finding those common hooks much easier to trace through the logic.

**Comment by eddyystop** [comment]
I suggest we create another hook that works like `disable` but with the correct sense. We can wean people away from `disable` by stressing the new hook in docs and maybe adding a deprecated message.

Any suggestions for the name of the replacement hook?

p.s. I've suggested something similar be done with the `remove` hook which is a no-op when on calls made by the server. I've suggested `delete` as the name of that replacement hook.
